paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/instrumentation/ResourcesCompatVisitorFactory.kt

```diff
@@ -0,0 +1,18 @@
+package app.cash.paparazzi.gradle.instrumentation
+
+import com.android.build.api.instrumentation.AsmClassVisitorFactory
+import com.android.build.api.instrumentation.ClassContext
+import com.android.build.api.instrumentation.ClassData
+import com.android.build.api.instrumentation.InstrumentationParameters
+import org.objectweb.asm.ClassVisitor
+
+internal abstract class ResourcesCompatVisitorFactory : AsmClassVisitorFactory<InstrumentationParameters.None> {
+  override fun createClassVisitor(
+    classContext: ClassContext,
+    nextClassVisitor: ClassVisitor
+  ): ClassVisitor {
+    return ResourcesCompatTransform(instrumentationContext.apiVersion.get(), nextClassVisitor)
+  }
+
+  override fun isInstrumentable(classData: ClassData): Boolean = true
+}
```

Comment:
nit: I rather like nesting the Transform class inside the factory class as done here: https://github.com/cashapp/paparazzi/pull/679/files#diff-b388a8009e690deb676684efd348e96131578944fae01677f60b93e921698a39

Shall we follow that pattern?  (keeping the name `ResourcesCompatTransform`)